src/webAppCreator.ts

```diff
@@ -11,16 +13,19 @@ import WebSiteManagementClient = require('azure-arm-website');
 import * as WebSiteModels from '../node_modules/azure-arm-website/lib/models';
 import * as util from './util';
 
-export class WebAppCreator extends WizardBase {
-    constructor(output: vscode.OutputChannel, readonly azureAccount: AzureAccountWrapper, subscription: SubscriptionModels.Subscription, persistence?: vscode.Memento) {
+export type WebsiteOS = "linux" | "windows";
+export type AppKind = "app" | "functionapp";
+
+export abstract class BaseWebsiteCreator extends WizardBase {
```

Comment:
nit: the naming is inconsistent with `WizardBase` and `WebsiteCreatorStepBase`

It should be `WebsiteCreatorBase` to match